.github/workflows/docker_snapshot.yml

```diff
@@ -0,0 +1,81 @@
+name: Build internal snapshot image
+
+on:
+  push:
+    branches:
+      - 'master'
+      - '[0-9]+.[0-9]+.z'
+
+jobs:
+  push:
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'hazelcast'
+    steps:
+      - name: Checkout hazelcast-snapshot repo
+        uses: actions/checkout@v2
+        with:
+          repository: hazelcast-dockerfiles/hazelcast-snapshot
+          ref: master
+          path: hazelcast-snapshot
+      - name: Checkout hazelcast-enterprise-snapshot repo
+        uses: actions/checkout@v2
+        with:
+          repository: hazelcast-dockerfiles/hazelcast-enterprise-snapshot
+          ref: master
+          path: hazelcast-enterprise-snapshot
+      - name: Checkout hazelcast repo
+        uses: actions/checkout@v2
+        with:
+          repository: hazelcast/hazelcast
+          path: hazelcast
+      - name: Checkout hazelcast-enterprise repo
+        uses: actions/checkout@v2
+        with:
+          repository: hazelcast/hazelcast-enterprise
+          ref: ${{ github.ref }}
+          path: hazelcast-enterprise
+          token: ${{ secrets.GH_PAT }}
+
+      - name: Set up Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to Docker Hub
+        run: echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin
```

Comment:
This will store hazelcast docker password unencrypted in GitHub servers during action run. I don't know how important it is.

Reply:
It's a good point. I believe the way is safe enough if we don't use untrusted actions. We already use this approach in other places - e.g. https://github.com/hazelcast/hazelcast-
docker/blob/v4.1.2/.github/workflows/oss_latest_snapshot_push.yml#L28-L29

The GitHub runners are ephemeral so the files should be destroyed after the run.

Still, I'll add to my TODO list to investigate a little bit around other approaches (e.g. 
https://docs.docker.com/engine/reference/commandline/login/#credentials-store)